observer/observer.go

```diff
@@ -0,0 +1,113 @@
+package observer
+
+import (
+	"context"
+	"errors"
+
+	errorsmod "cosmossdk.io/errors"
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	"github.com/cometbft/cometbft/libs/log"
+	"github.com/cometbft/cometbft/libs/pubsub/query"
+	rpchttp "github.com/cometbft/cometbft/rpc/client/http"
+	coretypes "github.com/cometbft/cometbft/rpc/core/types"
+	comettypes "github.com/cometbft/cometbft/types"
+)
+
+const ModuleNameObserver = "observer"
+
+type Observer struct {
+	logger        log.Logger
+	tmRpc         *rpchttp.HTTP
+	stopChan      chan struct{}
+	eventsOutChan chan abcitypes.Event
+}
+
+// NewObserver returns new instance of `Observer` with RPC client created
+func NewObserver(logger log.Logger, rpcUrl string) (Observer, error) {
+	if len(rpcUrl) == 0 {
+		return Observer{}, errors.New("RPC URL can't be empty")
+	}
```

Comment:
This check is redundant. `rpchttp.New` will return the `empty url` error in that case

Reply:
Yep, you are right, sorry for that. I understood the sources incorrectly